### PR TITLE
Pickupable inventory trigger

### PR DESCRIPTION
**Pull request by Cupelt**
# Add Command
`./trg i <trigger name> pickup`

> can Enable/Disable pickup Item in Inventory

### CODE REVIEW

**Comment by Cupelt** [comment]
Please ignore the previous commit lol;;

**Review by wysohn — CHANGES_REQUESTED** [review]
@Cupelt 
Please fix the test issues, and add specific test cases if possible

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTrigger.java [hunk 1]
```diff
@@ -82,6 +82,11 @@ public IItemStack[] getItems() {
         return items;
     }
 
+    public boolean isPickup() {
```
Comment (Sayakie) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTrigger.java:85]: This is good too, but I think it'd be better to `#canPickup()`.

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTriggerLoader.java [hunk 2]
```diff
@@ -84,8 +84,10 @@ public void save(InventoryTrigger trigger) {
 
             IItemStack[] items = trigger.getItems();
             int size = items.length;
+            boolean pickup = trigger.isPickup();
```
Comment (Sayakie) [inline core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTriggerLoader.java:87]: Same with the above thing.
```suggestion
            boolean pickup = trigger.isPickup();
            boolean canPickup = trigger.canPickup();
```

**Review by Sayakie — APPROVED** [review]
Just a bit things

**Review by Sayakie — CHANGES_REQUESTED** [review]
Oops, dind't mean approving...

File: core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTrigger.java (hunk 1 above)
Comment (Cupelt) [reply core/src/main/java/io/github/wysohn/triggerreactor/core/manager/trigger/inventory/InventoryTrigger.java:85]: Oh i think so

**Comment by Sayakie** [comment]
Nice work! It'd be cooler if has some tests belong to that to make sure do not break the consistent in some future.

File: core/src/test/java/io/github/wysohn/triggerreactor/core/main/TriggerReactorCoreTest.java [hunk 3]
```diff
@@ -936,6 +936,36 @@ public void command_invTrigger_item() throws Exception {
...
+        // act
+        handler.onCommand(sender, COMMAND_NAME, new String[]{"i", "MyInventory", "create", "54", "#MESSAGE \"Hello World\""});
+
+        // assert
+        assertNotNull(inventoryTriggerManager.get("MyInventory"));
+        assertEquals(54, inventoryTriggerManager.get("MyInventory").getItems().length);
+        assertEquals("#MESSAGE \"Hello World\"", inventoryTriggerManager.get("MyInventory").getScript());
+        assertEquals(false, inventoryTriggerManager.get("MyInventory").canPickup());
+
+        // act 2
```
Comment (wysohn) [inline core/src/test/java/io/github/wysohn/triggerreactor/core/main/TriggerReactorCoreTest.java:962]: I'd suggest separating `act 2` and `assert 2` into their own folder.

I know it's repetitive, but we could follow the best practices: https://medium.com/@tasdikrahman/f-i-r-s-t-principles-of-testing-1a497acda8d6

of course you can copy the arrange part

File: core/src/test/java/io/github/wysohn/triggerreactor/core/main/TriggerReactorCoreTest.java [hunk 4]
```diff
@@ -936,6 +936,36 @@ public void command_invTrigger_item() throws Exception {
...
+        when(sender.hasPermission(TRGCommandHandler.PERMISSION)).thenReturn(true);
+        when(pluginManagement.isEnabled()).thenReturn(true);
+
+        // act
+        handler.onCommand(sender, COMMAND_NAME, new String[]{"i", "MyInventory", "create", "54", "#MESSAGE \"Hello World\""});
+
+        // assert
+        assertNotNull(inventoryTriggerManager.get("MyInventory"));
+        assertEquals(54, inventoryTriggerManager.get("MyInventory").getItems().length);
+        assertEquals("#MESSAGE \"Hello World\"", inventoryTriggerManager.get("MyInventory").getScript());
```
Comment (Sayakie) [inline core/src/test/java/io/github/wysohn/triggerreactor/core/main/TriggerReactorCoreTest.java:959]: We can safely remove this parts as it is already tested in [#command_invTrigger_quick()]( https://github.com/TriggerReactor/TriggerReactor/pull/575/commits/251037b2abf94bda620df12dca7d4b9b44d55207#diff-fc2ca3016f00f5a6a916f837db86d4abbec7f44363917ed6b551e000e545de5dR831)

**Review by Sayakie — CHANGES_REQUESTED** [review]

**Review by wysohn — APPROVED** [review]
Looking good! Thanks a lot!

Code freeze on Sept 4th, so please merge before that day if you are ready

**Pull request merged**